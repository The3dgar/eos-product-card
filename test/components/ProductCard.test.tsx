import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { ProductCard } from '../../src';
import { product1 } from '../data/product';

describe('ProductCard', () => {
  test('should show component', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => <div>Custom content</div>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should increment the counter', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(+2)}>Increase</button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();
    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();
    expect((tree as any).children[1].children[0]).toBe('2');
  });
});
