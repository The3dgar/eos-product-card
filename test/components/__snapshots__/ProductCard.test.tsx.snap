// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should increment the counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
</div>
`;

exports[`ProductCard should show component 1`] = `
<div
  className="productCard undefined"
>
  <div>
    Custom content
  </div>
</div>
`;
